### ISSUE TITLE
`-(0:∞)` errors in Julia

### ISSUE THREAD

**Issue by KristofferC** [JuliaArrays/InfiniteArrays.jl#91]
```jl
julia> -(0:∞)
ERROR: ArgumentError: -1 is non-positive
Stacktrace:
 [1] *
   @ ~/.julia/packages/Infinities/WAhkk/src/cardinality.jl:139 [inlined]
 [2] range_start_step_length
   @ ./range.jl:204 [inlined]
 [3] _range
   @ ./range.jl:159 [inlined]
 [4] #range#67
   @ ./range.jl:138 [inlined]
 [5] -(r::InfiniteArrays.InfUnitRange{Int64})
   @ Base ./range.jl:1258
 [6] top-level scope
   @ REPL[10]:1
```

This line https://github.com/JuliaLang/julia/blob/62e0729dbc5f9d5d93d14dcd49457f02a0c6d3a7/base/range.jl#L204

fails with the inputs

```
1|julia> a, step, len, oneunit(len)
(0, -1, ℵ₀, 1)
```

**Comment by dlfivefifty** [comment]
Is this on 1.8? Im about to push fixes for 1.8 that may also resolve this.

**Comment by dlfivefifty** [comment]
Should be fixed
